rust/rust_c/src/trng.rs

```diff
@@ -0,0 +1,28 @@
+use core::num::NonZeroU32;
+
+use getrandom::{register_custom_getrandom, Error};
+
+use crate::bindings::GenerateTRNGRandomness;
+
+/// Same as in `keystore.h`.
+const ENTROPY_MAX_LEN: usize = 32;
+
+fn keystone_getrandom(dest: &mut [u8]) -> Result<(), Error> {
+    for chunk in dest.chunks_mut(ENTROPY_MAX_LEN) {
+        // SAFETY: `chunk.len()` is at most `ENTROPY_MAX_LEN` which fits in `u8`.
+        let len = chunk.len() as u8;
+
+        let ret = unsafe { GenerateTRNGRandomness(chunk.as_mut_ptr(), len) };
+
+        // TODO: Determine how to correctly map error codes from the underlying hardware
+        // into the `getrandom` custom error code space `Error::CUSTOM_START..=u32::MAX`.
```

Comment:
`Error::CUSTOM_START` happens to be 0xC0000000 so this mapping is not 1-1; all error codes `>= 0x3FFFFFFF` will be mapped to `Error::from(u32::MAX)`. It's probably fine.

Reply:
The problem is that the retcode type is `i32` so it can technically go below `Error::CUSTOM_START` if it is ever negative. My suspicion is that it won't actually go negative and this is just an artifact of C default types, but without access to the Secure Element's documentation I cannot know.

Reply:
I will do a errcode map later.